src/components/Lesson.js

```diff
@@ -57,12 +55,22 @@ const Lesson = ({ route, navigation, dispatch, completedGrammars }) => {
     }
   };
 
+  const showFooter = () => {
```

Comment:
I don't think you will need this function once you move `completedGrammars[grammar.id]` into a variable.